ddtrace/internal/agent.py

```diff
@@ -0,0 +1,30 @@
+import os
+
+from ddtrace.vendor import six
+from ddtrace import compat
+
+from .uds import UDSHTTPConnection
+
+
+HOSTNAME = os.environ.get("DD_AGENT_HOST", os.environ.get("DATADOG_TRACE_AGENT_HOSTNAME", "localhost"))
+PORT = int(os.environ.get("DD_AGENT_PORT", os.environ.get("DD_TRACE_AGENT_PORT", 8126)))
+URL = os.environ.get("DD_TRACE_AGENT_URL", "http://%s:%d" % (HOSTNAME, PORT))
+
+
+def get_connection(url, timeout):
+    if isinstance(url, six.string_types):
+        parsed = compat.parse.urlparse(url)
+    else:
+        parsed = url
```

Comment:
why? 😢 
just use one type of arg plz 😞

Reply:
Yeah it's not a good idea. I had done it to support not having to `compat.parse.urlparse` repeatedly but this might be a premature optimization.